lib/highscore.lua

```diff
@@ -75,11 +75,8 @@ function _hs_add(name, score)
   if index then
     _hs_last_index=index
 
-    for i=_hs_max_records,index+1,-1 do
-      _hs_records[i]=_hs_records[i-1]
-    end
-
-    _hs_records[index]={name=name,score=score}
+    add(_hs_records,{name=name,score=score},index)
+    _hs_records[_hs_max_records]=nil
```

Comment:
I think this should actually be nulling the record at the index after _hs_max_records. As it is now it will null out the 10th record instead.

```suggestion
    _hs_records[_hs_max_records+1]=nil
```